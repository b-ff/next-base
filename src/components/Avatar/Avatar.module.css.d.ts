declare const styles: {
  readonly "avatar": string;
  readonly "initials": string;
  readonly "image": string;
};
export = styles;

