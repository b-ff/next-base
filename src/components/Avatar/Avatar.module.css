.avatar {
  position: relative;
  border-radius: 50%;
  overflow: hidden;
  background-color: var(--color_border_primary);
  margin: 0;
  paddin: 0;
}

.initials {
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

.image {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 100%;
  height: 100%;
  object-fit: cover;
}
