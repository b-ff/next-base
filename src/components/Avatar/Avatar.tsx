"use client";
import { FC, HTMLProps, ReactNode } from "react";
import styles from "./Avatar.module.css";
import { classNames } from "@/utils";

const DEFAULT_SIZE = 24;

export type AvatarProps = HTMLProps<HTMLElement> & {
  src?: string;
  name: string;
  size?: number;
};

/**
 * Shows user profile picture
 */
export const Avatar: FC<AvatarProps> = (props) => {
  const {
    src = "",
    name = "",
    size = DEFAULT_SIZE,
    className = "",
    ...rest
  } = props;

  return (
    <figure
      className={classNames([styles.avatar, className])}
      style={{ width: size, height: size }}
      {...rest}
    >
      <div
        className={styles.initials}
        style={{ fontSize: Math.round(size * 0.4) }}
      >
        {name
          .toLocaleUpperCase()
          .split(" ")
          .map((word) => word[0])
          .join("")}
      </div>
      {Boolean(src) && <img src={src} alt={name} className={styles.image} />}
    </figure>
  );
};
