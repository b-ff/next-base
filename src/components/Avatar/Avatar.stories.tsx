import type { Meta, StoryObj } from "@storybook/react";

import { Avatar } from "./Avatar";

const meta: Meta<typeof Avatar> = {
  component: Avatar,
  tags: ["autodocs"],
};

export default meta;

type Story = StoryObj<typeof Avatar>;

export const Initials: Story = {
  args: {
    name: "John Doe",
  },
};

export const Image: Story = {
  args: {
    name: "Kermit The Frog",
    src: "https://lumiere-a.akamaihd.net/v1/images/character_themuppets_kermit_b77a431b.jpeg",
    size: 50,
  },
};

export const CustomSize: Story = {
  args: {
    name: "Kermit The Frog",
    src: "https://lumiere-a.akamaihd.net/v1/images/character_themuppets_kermit_b77a431b.jpeg",
    size: 100,
  },
};
