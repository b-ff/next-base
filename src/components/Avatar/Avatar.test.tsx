import { Avatar } from "./Avatar";
import { render } from "@testing-library/react";

const mockName = "John Doe";
const mockInitials = "JD";
const mockSrc =
  "https://lumiere-a.akamaihd.net/v1/images/character_themuppets_kermit_b77a431b.jpeg";

describe("Component Avatar", () => {
  it("Should render as expected", () => {
    const { container } = render(<Avatar name={mockName} />);
    expect(container).toMatchSnapshot();
  });

  it("Should render initials of given name", () => {
    const dom = render(<Avatar name={mockName} />);
    const initials = dom.getByText(mockInitials);
    expect(initials).not.toBeNull();
  });

  it("Should render image if src property was provided", () => {
    const { container } = render(<Avatar name={mockName} src={mockSrc} />);
    const img = container.querySelector("img");
    const src = img?.getAttribute("src");

    expect(img).not.toBeNull();
    expect(src).toBe(mockSrc);
  });
});
