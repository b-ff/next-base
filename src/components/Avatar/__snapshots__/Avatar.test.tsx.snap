// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Avatar Should render as expected 1`] = `
<div>
  <figure
    class="avatar"
    style="width: 24px; height: 24px;"
  >
    <div
      class="initials"
      style="font-size: 10px;"
    >
      JD
    </div>
  </figure>
</div>
`;
